setup.sh

```diff
@@ -103,694 +144,259 @@ vercomp () {
-          writeBold "$BLUE" "Ok."
-          agreeOnPersonalData="n"
-          echo
-          ;;
-        *)
-          writeBold "$RED" "[✖] Please, answer Y or N."
-          ;;
-      esac
-    done
-  else
-    writeBold "$BLUE" "Ok."
+    $KUZZLE_PUSH_ANALYTICS'{"type": "missing-docker", "uid": "'$ANALYTICS_UUID'", "os": "'$OS'"}' $ANALYTICS_URL &> /dev/null
+    ERROR=$MISSING_DEPENDENCY
+  fi
+
+  # Check if docker-compose is installed
+  if ! command_exists docker-compose; then
+    write_error "[✖] You need docker-compose to be able to run Kuzzle from this setup. Please install it and re-run this script"
+    write_error "    If you want to install Kuzzle without docker please see $INSTALL_KUZZLE_WITHOUT_DOCKER_URL"
     echo
-    purpose="Stop bugging me"
+    $KUZZLE_PUSH_ANALYTICS'{"type": "missing-docker-compose", "uid": "'$ANALYTICS_UUID'", "os": "'$OS'"}' $ANALYTICS_URL &> /dev/null    
+    ERROR=$MISSING_DEPENDENCY
   fi
-  $DL_BIN $UL_OPTS '{"type": "collected-data", "uid": "'$UUID'", "email": "'$email'", "name": "'"$name"'", "purpose": "'"$purpose"'", "os": "'$CURRENT_OS'"}' $ANALYTICS_URL &> /dev/null
-}
 
-startKuzzle() {
-  echo
-  if [ ! -f $COMPOSER_YML_PATH ]; then
-    $DL_BIN $UL_OPTS '{"type": "first-download", "uid": "'$UUID'"}' $ANALYTICS_URL &> /dev/null
+  # Check if docker version is at least $MIN_DOCKER_VER
+  if [ $ERROR -eq 0 ]; then
+    vercomp $(docker -v | sed 's/[^0-9.]*\([0-9.]*\).*/\1/') $MIN_DOCKER_VER
+    if [ $? -ne 0 ]; then
+      write_error "[✖] You need docker version to be at least $MIN_DOCKER_VER"
+      echo
+      $KUZZLE_PUSH_ANALYTICS'{"type": "docker-version-mismatch", "uid": "'$ANALYTICS_UUID'", "os": "'$OS'"}' $ANALYTICS_URL &> /dev/null    
+      ERROR=$MISSING_DEPENDENCY
+    fi
   fi
 
-  if [ -z $DL_BIN ]; then
-    failDL
+  # Check if sysctl exists on the machine
+  if ! command_exists sysctl; then
+    write_error "[✖] This script needs sysctl to check whether your kernel settings fit the requirements of Kuzzle."
+    write_error "    Please install sysctl and re-run this script."
+    echo
+    $KUZZLE_PUSH_ANALYTICS'{"type": "missing-sysctl", "uid": "'$ANALYTICS_UUID'", "os": "'$OS'"}' $ANALYTICS_URL &> /dev/null    
+    ERROR=$MISSING_DEPENDENCY
   else
-    writeBold "[ℹ] Downloading Kuzzle launch file..."
-    $DL_BIN $DL_OPTS $COMPOSER_YML_URL > $COMPOSER_YML_PATH
+    # Check of vm.max_map_count is at least $MIN_MAX_MAP_COUNT
+    VM_MAX_MAP_COUNT=$(sysctl -n vm.max_map_count)
+    if [ -z "${VM_MAX_MAP_COUNT}" ] || [ ${VM_MAX_MAP_COUNT} -lt $MIN_MAX_MAP_COUNT ]; then
+      write_error
+      write_error "[✖] The current value of the kernel configuration variable vm.max_map_count (${VM_MAX_MAP_COUNT})"
+      write_error "    is lower than the required one ($MIN_MAX_MAP_COUNT+)."
+      write_error "    In order to make ElasticSearch working please set it by using on root: (more at https://www.elastic.co/guide/en/elasticsearch/reference/5.x/vm-max-map-count.html)"
```

Comment:
`In order to make Elasticsearch work, you need to set it with the following command (needs root access):`
(note how `Elasticsearch`'s official name is not camelcased: https://www.elastic.co/webinars/getting-started-elasticsearch)